dask/async.py

```diff
@@ -488,12 +488,10 @@ def fire_task():
                 task = dsk[key]
                 _execute_task(task, data)  # Re-execute locally
             else:
-                raise type(res)('\nRemote Exception:\n'
-                              + '-----------------\n'
-                              + str(res) + '\n\n'
-                              + 'Traceback:\n'
-                              + '----------\n'
-                              + tb)
+                print("Traceback:\n"
+                      "----------")
+                print(tb)
```

Comment:
These print statements can't be caught, meaning that if the user catches the exception they'll still be displayed. What about just creating a new exception class to represent all remote errors?

```python
class RemoteError(StandardError):
    def __init__(self, err, tb):
        self.err = err
        self.tb = tb
        StandardError.__init__(self, (err, tb))

    def __str__(self):
        return ('\n\nOriginal Error:\n'
                '---------------\n'
                '{cls}: {err}\n\n'
                'Traceback:\n'
                '----------\n'
                '{tb}\n').format(cls=type(self.err).__name__,
                                 err=self.err, tb=self.tb)
```

In use:

```python
In [109]: e
Out[109]: UnicodeEncodeError('charmap', u'\u0411', 0, 1, 'character maps to <undefined>')

In [110]: r = RemoteError(e, 'This is a traceback\n woooooo')

In [111]: raise r
---------------------------------------------------------------------------
RemoteError                               Traceback (most recent call last)
<ipython-input-111-252858239410> in <module>()
----> 1 raise r

RemoteError:

Original Error:
---------------
UnicodeEncodeError: 'charmap' codec can't encode character u'u0411' in position 0: character maps to <undefined>

Traceback:
----------
This is a traceback
 woooooo
```

I like this because:

- Simple. Tell user: scheduler raises `RemoteError`, which has `err` and `tb` attributes for accessing original error(could make those longer names)
- Doesn't involve mucking about in modifying exceptions
- No `print` statements in code.